Doc/library/types.rst

```diff
@@ -262,11 +262,22 @@ Standard names are defined for the following types:
 
    .. versionadded:: 3.10
 
-.. data:: GenericAlias
+.. class:: GenericAlias(t_origin, t_args)
 
    The type of :ref:`parameterized generics <types-genericalias>` such as
    ``list[int]``.
 
+   ``t_origin`` should be a non-parameterized generic class, such as ``list``,
+   ``tuple`` or ``dict``.  ``t_args`` should be a :class:`tuple` (possibly of
+   length 1) of types which parameterize ``t_origin``::
```

Comment:
The argument types are not checked, e.g. `types.GenericAlias(1, 2)` returns something with `__origin__ == 1` and `__args__ == (2,)`. But I'm not sure we ought to document that -- perhaps it's better to leave that out (even if I can see some odd use cases for it).

Reply:
Yeah I noticed that you can produce things like `int[str]` as well (or anything for that matter). IMO, it would be best to document the current expected/common/intended use case for it. In the future if `GenericAlias` is used for more functionality, someone can just document that.